File: locust_benchmark/benchmark_log.py
import time
import warnings
from locust import HttpUser, task, between

warnings.filterwarnings("ignore")

class QuickstartUser(HttpUser):
    wait_time = between(1, 2.5)

    @task
    def log(self):
        self.client.get("/api/log", verify=False)

    @task
    def log_with_params(self):
        self.client.post("/api/log", verify=False, json={
            "quantity": 1
        })

    @task
    def log_data(self):
        self.client.post("/api/data", verify=False, json={
            "database": "log"
        })

    @task
    def log_data_with_quantity(self):
        self.client.post("/api/data", verify=False, json={
            "database": "log",
            "quantity": 50
        })

File: k6_benchmark/benchmark_log.js
import http from 'k6/http';
import { sleep } from 'k6';

export const options = {
    duration: '1m',
    vus: 150,
    insecureSkipTLSVerify: false,
};

export default function () {  
    http.post('https://localhost/api/log', {quantity: 1});
    http.get('https://localhost/api/log');

    http.post('https://localhost/api/data', {database: "log"})
    http.post('https://localhost/api/data', {database: "log", quantity: 50})
    sleep(1);
}